app/src/main/java/com/es/findsoccerplayers/fragments/FragmentBookedMatches.java

```diff
@@ -27,27 +30,40 @@
 
     private FirebaseDatabase db = FirebaseDatabase.getInstance();
     private MatchAdapter matchAdapter;
+    RecyclerView recyclerView;
+    LinearLayoutManager linearLayoutManager;
     private List<Match> matches = new ArrayList<>();
 
     @Override
     public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
         View view = inflater.inflate(R.layout.frag_booked_matches, container, false);
 
-        RecyclerView recyclerView = view.findViewById(R.id.frag_booked_list);
+        recyclerView = view.findViewById(R.id.frag_booked_list);
         recyclerView.setHasFixedSize(true);
-        LinearLayoutManager linearLayoutManager = new LinearLayoutManager(getContext());
-        recyclerView.setLayoutManager(linearLayoutManager);
+        linearLayoutManager = new LinearLayoutManager(getContext());
+        matchAdapter = new MatchAdapter(matches);
 
-        matchAdapter = new MatchAdapter(getContext(), matches);
+        recyclerView.setLayoutManager(linearLayoutManager);
+        recyclerView.addItemDecoration(new DividerItemDecoration(recyclerView.getContext(), DividerItemDecoration.VERTICAL));
         recyclerView.setAdapter(matchAdapter);
         recyclerView.setItemAnimator(null);
+
+        matchAdapter.setOnItemClickListener(new MatchAdapter.OnItemClickListener() {
+            @Override
+            public void onItemClick(int position) {
+                matches.get(position);
+                Intent i = new Intent(getContext(), ActivityInfoBookedMatch.class);
+                i.putExtra("match", matches.get(position).getMatchID());
+                i.setFlags(Intent.FLAG_ACTIVITY_REORDER_TO_FRONT);
```

Comment:
perché il flag?

Reply:
Non è il flag per tenere in memoria l'activity senza che venga ricreata?

Reply:
no quel flag serve a startare l'activity già presente nello stack